### PR TITLE
Webfinger resource to extract username from resource string

### PR DESCRIPTION
**Pull request by mjankowski**
I noticed the logic in this controller contained a reference to `params_path[:controller] == 'users'` which I think must be for outdated code (that route no longer exists?). I extracted the logic to pull out a username from a resource to it's own class, and added specs for the strings I'm guessing this is supposed to handle.

Feedback on whether that list of strings is exhaustive and sufficient would be appreciated!

### CODE REVIEW

File: app/lib/webfinger_resource.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class WebfingerResource
+  attr_reader :resource
+
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def username
+    case resource
+    when /https/
```
Comment (Gargron) [inline app/lib/webfinger_resource.rb]: I'm thinking this should be `/https?/`

Comment (mjankowski) [inline app/lib/webfinger_resource.rb]: Ah I missed that in the original ... so both of these strings should work?

- `http://example.com/users/name` -> `name`
- `https://example.com/users/name` -> `name`

?

Comment (Gargron) [inline app/lib/webfinger_resource.rb]: Although your tests seem to account for the http case so I'm not sure what this regex does

Comment (mjankowski) [inline app/lib/webfinger_resource.rb]: PR updated to reflect that https is supported.

File: app/lib/webfinger_resource.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  def initialize(resource)
+    @resource = resource
+  end
+
+  def username
+    case resource
+    when /\Ahttps?/i
+      username_from_url
+    when /acct/
```
Comment (Gargron) [inline app/lib/webfinger_resource.rb]: Thinking right now - You should be able to pass username@domain without the acct: in the front, and it should be able to deal with that. So maybe just make that the else condition.

Comment (Gargron) [inline app/lib/webfinger_resource.rb]: I should have submitted that as a change request - this should be able to handle `username@domain` without the `acct:` at the front

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by mjankowski** [comment]
Updated to handle the simple `user@host` (w/out `acct:`) as well.

**Review by Gargron — APPROVED** [review]

**Pull request merged**